crates/bevy_ecs/src/lib.rs

```diff
@@ -171,8 +173,10 @@ mod tests {
         }
     }
 
-    // TODO: The compiler says the Debug and Clone are removed during dead code analysis. Investigate.
-    #[expect(dead_code)]
+    #[expect(
+        dead_code,
+        reason = "This struct is used to test how `Drop` behavior works in regards to SparseSet storage, and as such is solely a wrapper around `DropCk` to make it use the SparseSet storage. Because of this, the inner field is intentionally never read."
```

Comment:
I'm particularly proud of this comment. It took quite a bit to figure out the struct's use, and longer still to nail the messaging, but I'm proud of it.